### PR TITLE
Include dotnet-watch redist output in layouts

### LINKED ISSUE #29609: dotnet-watch tool missing in SDK
The `dotnet watch` tool is missing from the .NET 8 SDK. This repros on build 8.0.100-alpha.1.22619.1.

On a machine with .NET 8 SDK installed, you get this:

```
/ # dotnet watch
Cannot find command 'dotnet watch', run the following command to install

dotnet tool install --global dotnet-watch
```

You can easily verify this by extracting https://dotnetbuilds.azureedge.net/public/Sdk/8.0.100-alpha.1.22619.1/dotnet-sdk-8.0.100-alpha.1.22619.1-win-x64.zip and see that `sdk\8.0.100-alpha.1.22619.1\DotnetTools\dotnet-watch` is missing.

Possibly caused by https://github.com/dotnet/sdk/pull/29343? cc @tmat

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/sdk/issues/29609

### CODE REVIEW

File: src/Tests/dotnet-watch.Tests/dotnet-watch.Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,4 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-  <Import Project="$(RepoRoot)\src\Layout\redist\targets\PublishDotnetWatch.targets"/>
```
Comment (tmat) [inline src/Tests/dotnet-watch.Tests/dotnet-watch.Tests.csproj:2]: Moved to `dotnet-watch.csproj`

**Comment by tmat** [comment]
@dsplaisted @arkalyanms  PTAL

**Comment by tmat** [comment]
@dsplaisted @arkalyanms Ping

**Comment by mkArtakMSFT** [comment]
@dsplaisted , @arkalyanms looks like this PR is awaiting your review. We're getting close to 8.0 preview1 release and there are several bugs blocked on this PR. Could you please prioritize reviewing this so that @tmat can get this merged? Thanks!

**Review by dsplaisted — APPROVED** [review]
Looks good.  Did you verify that dotnet-watch is now included in the dotnet-toolset-internal zip that is created?  That's what gets consumed in dotnet/installer, but the CI build has expired so I can't check it in the CI build artifacts myself.

**Comment by tmat** [comment]
I'll run build again and check.

**Comment by tmat** [comment]
@dsplaisted 
It's there: `dotnet-toolset-internal-8.0.100-ci.zip\DotnetTools\dotnet-watch\8.0.100-ci\tools\net8.0\any`

**Pull request merged**